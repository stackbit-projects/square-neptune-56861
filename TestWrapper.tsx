import React from 'react';
import { ThemeProvider } from "styled-components";
import { theme } from './theme';

export const TestWrapper = (props) => {
 return (
  <ThemeProvider theme={theme}>
    {props.children}
  </ThemeProvider>
 )
}
