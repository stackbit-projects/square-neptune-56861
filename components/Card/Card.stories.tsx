import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Card from '.';

export default {
  title: 'Card | ManualNavigationPod',
  component: Card,
} as Meta;

const Template: Story = (args) => <Card  {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        title: "Lorem ipsum dolor sit amet",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ac lectus et velit vulputate elementum. Aenean euismod tortor id sapien accumsan, vel commodo metus luctus. Mauris dapibus consectetur sem quis ultricies. Nam iaculis velit in tortor lobortis, in egestas augue facilisis. Fusce mattis pulvinar felis nec vulputate. Sed vulputate sem id nisi cursus, in ornare erat eleifend. Aliquam ut velit volutpat, volutpat nisi in, molestie nulla.",
        "pod image": {
          url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
          alt: "cute monkey"
        },
        link: {
          url: "/",
          text: "button text"
        }
      }
    }
  }
};

export const Secondary = Template.bind({});
Secondary.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        title: "Component without link data",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ac lectus et velit vulputate elementum. Aenean euismod tortor id sapien accumsan, vel commodo metus luctus. Mauris dapibus consectetur sem quis ultricies. Nam iaculis velit in tortor lobortis, in egestas augue facilisis. Fusce mattis pulvinar felis nec vulputate. Sed vulputate sem id nisi cursus, in ornare erat eleifend. Aliquam ut velit volutpat, volutpat nisi in, molestie nulla.",
        "pod image": {
          url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
          alt: "cute monkey"
        },
      }
    }
  }
};
export const Tertiary = Template.bind({});
Tertiary.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        title: "Component without image data",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ac lectus et velit vulputate elementum. Aenean euismod tortor id sapien accumsan, vel commodo metus luctus. Mauris dapibus consectetur sem quis ultricies. Nam iaculis velit in tortor lobortis, in egestas augue facilisis. Fusce mattis pulvinar felis nec vulputate. Sed vulputate sem id nisi cursus, in ornare erat eleifend. Aliquam ut velit volutpat, volutpat nisi in, molestie nulla.",
        link: {
          url: "/",
          text: "button text"
        }
      }
    }
  }
};