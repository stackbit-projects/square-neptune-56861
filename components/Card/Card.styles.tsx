import styled from 'styled-components'

export const Root = styled.div`
  width: 100%;
  margin: .75rem auto;
  padding: 0;
  ${props => props.theme.breakpoints.sm} {
    margin: 1rem auto;
  }
  ${props => props.theme.breakpoints.md} {
    margin: 1.5rem auto;
  }
`;

export const Wrapper = styled.div`
  background: white;
  margin: 0;
  padding: 0;
  min-height: 6.25rem;
  -webkit-transition: min-height .25s ease-in-out;
  transition: min-height .25s ease-in-out;
  word-wrap: break-word;
  overflow-wrap: break-word;
  border-radius: .5rem;
  overflow: hidden;
  border: 2px solid #ebebeb;
`;

export const Content = styled.div`
  padding: 1.5rem;
  ${props => props.theme.breakpoints.md} {
    padding: 2rem;
  }
`

export const Text = styled.p`
  margin: 1rem 0 0 0;
  display: block;
`

export const ImageWrapper = styled.div<{isLarge: boolean}>`
  padding-bottom: ${props => props.isLarge ? '75%' : '56.25%'};
  overflow: hidden;
  display: block;
  height: 0;
  position: relative;
`

export const Image = styled.img`
  height: 100%;
  width: 100%;
  object-fit: cover;
  font-family: 'object-fit: cover;';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`

export const Link = styled.a`
  font-size: 1.625rem;
  line-height: 2.25rem;
  text-decoration: underline;
`