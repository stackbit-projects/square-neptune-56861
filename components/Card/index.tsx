import React from 'react'
import { linkFormatter } from '../../utils/formatter'
import { getBlueBackground } from '../../utils/styleClass'
import { Root, Wrapper, Content, Text, ImageWrapper, Image, Link } from "./Card.styles"

const Variants = {
  Default: "{D288AE53-0DFF-418F-9773-52A51A40B582}",
  NoImage: "{83D702CA-4797-4B98-9C9A-E861370CE490}",
  Large: "{0199A687-14BF-4599-A99A-6A97576E18D8}"
}

const Styles = {
  MakeEqualHeight: "{3182392E-93CD-425C-8778-611B677FA58A}",
}

const Card = (props: any) => {
  const { renderingContext } = props

  if (!renderingContext.item || !renderingContext.item.fields) {
    console.error("No item supplied for Card")
    return null
  }

  const { item: { id, name, fields }, parameters } = renderingContext

  const isLarge = parameters?.FieldNames === Variants.Large;
  const noImage = parameters?.FieldNames === Variants.NoImage;
  const jsClass = parameters?.Styles === Styles.MakeEqualHeight;
  const backgroundStyle = getBlueBackground(parameters?.Styles)

  let formattedUrl = fields.link?.url.split("/") ?? []
  if (formattedUrl.length > 0) {
    formattedUrl = formattedUrl[formattedUrl.length - 1].replace(/-/g, " ")
  } else {
    formattedUrl = ""
  }

  return (
    <Root className={`component c-navigationPod c-navigationPod--manual small-12 columns ${backgroundStyle} ${jsClass ? "js-equalHeight" : ""}`}>
      <div className="component-content">
        <Wrapper className="c-navigationPod__wrapper">
          {!noImage && fields['pod image'] &&
            <ImageWrapper isLarge={isLarge}>
              {fields['pod image'].alt 
               ? <Image src={fields['pod image'].url} alt={fields['pod image'].alt} loading="lazy" />
               : <Image src={fields['pod image'].url} loading="lazy" />
              }
            </ImageWrapper>
          }
          <Content>
            {fields.link &&
              <div className=" field-link">
                <Link href={linkFormatter(fields.link)} data-variantitemid={`{${id}}`} data-variantfieldname="Link">{fields.link.text || formattedUrl}</Link>
              </div>
            }
            <Text className="field-pod-text" dangerouslySetInnerHTML={{ __html: fields['pod text'] }} />
            {isLarge && fields['cta link'] &&
              <div className="c-navigationPod__cta field-cta-link">
                <a href={linkFormatter(fields['cta link'])} data-variantitemid={`{${id.toUpperCase()}}`} data-variantfieldname="CTA link">{fields['cta link'].text}</a>
              </div>
            }
          </Content>
        </Wrapper>
      </div>
    </Root>
  )
}

export default Card