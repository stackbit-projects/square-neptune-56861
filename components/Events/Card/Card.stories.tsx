import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import EventCard from '.';

export default {
  title: 'Event Card',
  component: EventCard,
} as Meta;

const Template: Story = (args) => <EventCard  {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        status: "",
        title: "London Landmark Half Marathon",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur efficitur pulvinar malesuada. Mauris dapibus tortor sed iaculis consequat. Curabitur iaculis massa ac.",
        "pod image": {
          url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
          alt: "cute monkey"
        },
        link: {
          url: "/",
        }
      }
    }
  }
};

export const Full = Template.bind({});
Full.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        status: "full",
        title: "London Landmark Half Marathon",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur efficitur pulvinar malesuada. Mauris dapibus tortor sed iaculis consequat. Curabitur iaculis massa ac.",
        "pod image": {
          url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
          alt: "cute monkey"
        },
        link: {
          url: "/",
        }
      }
    }
  }
};

export const Cancelled = Template.bind({});
Cancelled.args = {
  renderingContext: {
    id: '1234',
    item: {
      fields: {
        status: "cancelled",
        title: "London Landmark Half Marathon",
        "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur efficitur pulvinar malesuada. Mauris dapibus tortor sed iaculis consequat. Curabitur iaculis massa ac.",
        "pod image": {
          url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
          alt: "cute monkey"
        },
        link: {
          url: "/",
        }
      }
    }
  }
};
