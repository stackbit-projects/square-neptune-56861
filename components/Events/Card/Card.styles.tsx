import styled from 'styled-components'
import { typography } from '../../../theme'

export const Root = styled.div`
  position: relative;
  border: 2px solid ${props => props.theme.colors.grey};
  border-radius: 6px;
  background-color: ${props => props.theme.colors.white};
`

export const ImageWrapper = styled.div<{status: "full" | "cancelled"}>`
  position: relative;
  &::after {
    background: ${props => props.status === 'full' ? props.theme.colors.secondary : props.theme.colors.error};
    opacity: 0.4;
    position: absolute;
    display: block;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    content: "";
  }
`
export const Image = styled.img`
  height: 100%;
  width: 100%;
  display: block;
  object-fit: cover;
 
`

export const Wrapper = styled.div`
  padding: 32px 32px 68px;
`

export const Banner = styled.div<{status: "full" | "cancelled"}>`
  ${typography.h5}
  background: ${props => props.status === 'full' ? props.theme.colors.secondary : props.theme.colors.error};
  color: ${props => props.status === 'full' ? props.theme.colors.primary : props.theme.colors.white};
  padding: 6px 25px;
`

export const Title = styled.h3`
  ${typography.h3}
  color: ${props => props.theme.colors.primary};
  a {
    text-decoration: none;
    border: none;
    ${typography.h3}
    &:hover {
      background: none;
      color: inherit;
    }
  }
`

export const Summary = styled.p`
  ${typography.body}
  color: ${props => props.theme.colors.primary};
  margin: 0;
`