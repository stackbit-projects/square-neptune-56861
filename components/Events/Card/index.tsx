import React from 'react'
import { Root, Image, Title, Summary, Wrapper, Banner, ImageWrapper } from './Card.styles'

const EventCard = (props) => {
  const { item: { fields } } = props.renderingContext

  let bannerText = ""
  if (fields.status === "full") {
    bannerText = "Event capacity full"
  } else if (fields.status === "cancelled") {
    bannerText = "Event cancelled"
  }
  return (
    <React.Fragment>
      <Root>
        <ImageWrapper status={fields.status}>
          <Image src={fields["pod image"].url} alt={fields["pod image"].alt} />
        </ImageWrapper>
        {bannerText && <Banner status={fields.status}>{bannerText}</Banner>}
        <Wrapper>
          <Title><a href={fields.link.url}>{fields.title}</a></Title>
          <Summary>{fields["pod text"]}</Summary>
        </Wrapper>
      </Root>
    </React.Fragment>
  )
}

export default EventCard