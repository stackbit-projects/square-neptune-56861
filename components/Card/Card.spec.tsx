import React from 'react';
import renderer from 'react-test-renderer';
import Card from '.';
import { TestWrapper } from '../../TestWrapper';

it('Card renders correctly', () => {
  const tree = renderer
    .create(<TestWrapper>
      <Card renderingContext={{
        page: {
          renderings: [{
            settings: {
              DataSource: ''
            }
          }]
        },
        item: {
          id: '123',
          fields: {
            title: "Lorem ipsum dolor sit amet",
            "pod text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ac lectus et velit vulputate elementum. Aenean euismod tortor id sapien accumsan, vel commodo metus luctus. Mauris dapibus consectetur sem quis ultricies. Nam iaculis velit in tortor lobortis, in egestas augue facilisis. Fusce mattis pulvinar felis nec vulputate. Sed vulputate sem id nisi cursus, in ornare erat eleifend. Aliquam ut velit volutpat, volutpat nisi in, molestie nulla.",
            "pod image": {
              url: "https://img.posterlounge.co.uk/images/big/1592909.jpg",
              alt: "cute monkey"
            },
            link: {
              url: "/",
              text: "button text"
            }
          }
        }
      }} />
      </TestWrapper>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});