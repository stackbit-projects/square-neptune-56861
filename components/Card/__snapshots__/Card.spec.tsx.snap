// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-bdfBwQ fzlwWY component c-navigationPod c-navigationPod--manual small-12 columns  "
>
  <div
    className="component-content"
  >
    <div
      className="sc-gsTCUz lgbzsf c-navigationPod__wrapper"
    >
      <div
        className="sc-eCssSg flaySL"
      >
        <img
          alt="cute monkey"
          className="sc-jSgupP gbrltj"
          loading="lazy"
          src="https://img.posterlounge.co.uk/images/big/1592909.jpg"
        />
      </div>
      <div
        className="sc-dlfnbm edcLZW"
      >
        <div
          className=" field-link"
        >
          <a
            className="sc-gKsewC cKBQWx"
            data-variantfieldname="Link"
            data-variantitemid="{123}"
            href="/"
          >
            button text
          </a>
        </div>
        <p
          className="sc-hKgILt fZHvtl field-pod-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris ac lectus et velit vulputate elementum. Aenean euismod tortor id sapien accumsan, vel commodo metus luctus. Mauris dapibus consectetur sem quis ultricies. Nam iaculis velit in tortor lobortis, in egestas augue facilisis. Fusce mattis pulvinar felis nec vulputate. Sed vulputate sem id nisi cursus, in ornare erat eleifend. Aliquam ut velit volutpat, volutpat nisi in, molestie nulla.",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
